tests/PilotLive/ValidationTest.php

```diff
@@ -0,0 +1,246 @@
+            'Validation error value  failed to validate filter_var failed to validate',
+            ValidationException::FILTER_VAR_FAILED
+        );
+        $this->validateFilterVar(false, FILTER_VALIDATE_BOOLEAN);
+    }
+
+
+    //above tests reach 100% coverage, robustness tests below
+    public function testValidateNull()
+    {
+        $this->assertException(
+            ValidationException::class,
+            'Validation error value  is type NULL Validation type is invalid',
+            ValidationException::VALIDATION_TYPE_ERROR
+        );
+        $this->validateRange(null, 5, 15);
+    }
+    public function testValidateNullType()
+    {
+        //Should this be valid?
+        $this->assertTrue($this->isValidPrimitive("\0foo", 'string'));
+    }
+    public function testValidateDifferentNullType()
+    {
+        //Should this be valid?
+        $this->assertTrue($this->isValidPrimitive("\x00foo", 'string'));
+    }
+    public function testValidateURLHexEncodedType()
+    {
+        //URLEncoding
+        //Should this be valid?
+        $this->assertTrue($this->isValidPrimitive("foo%60%7ebar", 'string'));
+        //Should this be valid?
+        $this->assertTrue($this->isValidPrimitive(urldecode("foo%60%7ebar"), 'string'));
+        //HexEncoding
+        $this->assertTrue($this->isValidPrimitive(bin2hex("foobar"), 'string'));
+        //Validate range is moved to bottom test
+    }
+    public function testValidateBaseSixtyFourEncodedType()
+    {
+        //b64 encoding tests
+        $this->assertTrue($this->isValidPrimitive(base64_encode("foobar"), 'string'));
+        $this->validateRange(base64_encode("foobar"), 5, 15);
+    }
+    public function testEmojiValidation()
+    {
+        //is this valid?
+        $this->validateRange(json_decode('"\uD83D\uDE00"'), 0, 8);
+    }
+
+    public function testOutlierChars()
+    {
+        $this->validateRange(0x00, 0, 8);
+        $this->assertException(
+            ValidationException::class,
+            "Validation error value 255 out of min(0) max(8) Integer value is out of range",
+            ValidationException::INTEGER_OUT_OF_RANGE
+        );
+        $this->validateRange(0xFF, 0, 8);
+    }
+    private function assertException($class, String $message, int $code)
+    {
+        $this->expectException($class);
+        $this->expectExceptionMessage($message);
+        $this->expectExceptionCode($code);
+    }
+    /**
+     * @dataProvider stringExceptionProvider
+     */
+    public function testLengthValidation($message, $test, $min, $max)
+    {
+        $this->assertException(
+            ValidationException::class,
+            sprintf("Validation error value %s String length is out of range", $message),
+            ValidationException::STRING_LENGTH_OUT_OF_RANGE
+        );
+        $this->validateRange($test, $min, $max);
+    }
+    public static function stringExceptionProvider()
+    {
+        return array(
+            array(
```

Comment:
You should use short array syntax always.

These can be replaced with `[]` notation.

Reply:
Fixed 👍